File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-0.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int right = 0 ;
        int left = 0 ;
        set<char> previousChar;
        int maxLength = 0 ;
        for(right = 0 ; right < s.size() ; right++){   
                   
            while(previousChar.find(s[right])!=previousChar.end()){
                previousChar.erase(s[left]);
                left++;
            }
            cout<<s[right]<<endl;
            maxLength = max(maxLength,right - left + 1); 
            previousChar.insert(s[right]);            
        }
        return maxLength;
    }
};

File: Data Structures & Algorithms/binary-tree-diameter/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    res = 0
    def diameterOfBinaryTree(self, root: Optional[TreeNode]) -> int:        
        self.dfs(root)
        return self.res
    def dfs(self,root):        
        if root is None :
            return 0
        
        left = self.dfs(root.left) 
        right = self.dfs(root.right) 
        print(left + right)
        self.res = max(self.res, left + right)
        return (max(left,right) + 1)
    
    
    
        